eth/vm/logic/context.py

```diff
@@ -76,6 +76,10 @@ def calldatacopy(computation: BaseComputation) -> None:
     computation.memory_write(mem_start_position, size, padded_value)
 
 
+def chain_id(computation: BaseComputation) -> None:
+    computation.stack_push(computation.transaction_context.chain_id)
```

Comment:
This line needs to change to not use the transaction context value

Reply:
Where do you foresee the `chain_id` being retrieved from?

Reply:
The `BaseChain` class and all subclasses define the correct configuration we wish to use.

Unfortunately, using the transaction context would not be valid because technically it is still valid to forgo using EIP-155 when signing a transaction, and we cannot have the opcode returning a different value based on the message signer's choice as it could be a security bug where the L2 message signer abd the L1 message signer disagree, and the messages are considered invalid when they were correctly signed.

Reply:
Wherever it gets stored, we should probably ensure we have some rigorous tests for this in place to be extra sure that we don't end up leaking the wrong value.

I'm thinking that this might justify a different data storage object, something like `ChainContext` that might only house this value for now but would give us an appropriate place to store it.

Reply:
Still WIP, but I added that `ChainContext` class to `BaseVM`/`VM`

Unsure of how to pass it to `BaseVM` classes though...